axon-server-connector/src/main/java/org/axonframework/axonserver/connector/command/AxonServerCommandBusConnector.java

```diff
@@ -0,0 +1,297 @@
+                        "Timed out waiting for subscription acknowledgment for command: " + commandName, e
+                );
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+                throw new RuntimeException("Thread interrupted while waiting for subscription acknowledgment", e);
+            }
+        }
+        this.subscriptions.put(commandName, registration);
+    }
+
+    private CompletableFuture<CommandResponse> incoming(Command command) {
+        logger.info("Received incoming command [{}]", command.getName());
+        try {
+            CompletableFuture<CommandResponse> result = new CompletableFuture<>();
+            incomingHandler.get().handle(convertToCommandMessage(command),
+                                         new FutureResultCallback(result, command));
+            return result;
+        } catch (Exception e) {
+            logger.error("Error processing incoming command: {}", command.getName(), e);
+            CompletableFuture<CommandResponse> errorResult = new CompletableFuture<>();
+            errorResult.completeExceptionally(e);
+            return errorResult;
+        }
+    }
+
+    private CommandMessage<?> convertToCommandMessage(Command command) {
+        SerializedObject commandPayload = command.getPayload();
+        int priority = priority(command.getProcessingInstructionsList());
+        String routingKey = ProcessingInstructionHelper.routingKey(command.getProcessingInstructionsList());
+        return new GenericCommandMessage<>(
+                new GenericMessage<>(
+                        command.getMessageIdentifier(),
+                        new MessageType(commandPayload.getType(), commandPayload.getRevision()),
+                        commandPayload.getData().toByteArray(),
+                        convertFromGrpcMetaDataValues(command.getMetaDataMap())
+                ),
+                routingKey,
+                priority
+        );
+    }
+
+    private CommandResponse createResult(Command command, Message<?> resultMessage) {
+        if (resultMessage == null) {
+            return CommandResponse.newBuilder()
+                                  .setMessageIdentifier(UUID.randomUUID().toString())
+                                  .build();
+        }
+
+        String messageId = getOrDefault(resultMessage.getIdentifier(), UUID.randomUUID().toString());
+        CommandResponse.Builder responseBuilder = CommandResponse
+                .newBuilder()
+                .setMessageIdentifier(messageId)
+                .putAllMetaData(AxonServerMetadataConverter.convertToGrpcMetaDataValues(resultMessage.getMetaData()))
+                .setRequestIdentifier(command.getMessageIdentifier());
+        if (!(resultMessage.getPayload() instanceof byte[] payloadAsBytes)) {
+            throw new IllegalArgumentException(
+                    "Payload must be of type byte[] for AxonServerConnector, but was: %s".formatted(
```

Comment:
Adding the same conversion-issue mention as with the previous `IllegalArgumentException`